packages/components/src/components/segmented-control/segmented-control-item/segmented-control-item-styles.ts

```diff
@@ -21,38 +24,36 @@ export const ICON_MARGIN = pxToRemWithUnit(4);
 export const getColors = (
   isDisabled: boolean,
   isSelected: boolean,
-  bgColor: SegmentedControlBackgroundColor,
   theme: Theme
-): { backgroundColor: string; buttonColor: string; labelColor: string } => {
+): {
+  buttonColor: string;
+  labelColor: string;
+  borderColor: string;
+  hoverBorderColor: string;
+} => {
   const themedColors = getThemedColors(theme);
-  const { primaryColor, contrastMediumColor } = isSelected ? getInvertedThemedColors(theme) : themedColors;
 
-  const backgroundColor =
-    themedColors[
-      isSelected ? 'contrastHighColor' : bgColor === 'background-surface' ? 'backgroundColor' : 'backgroundSurfaceColor'
-    ];
+  const { primaryColor, contrastMediumColor, disabledColor, contrastLowColor } = themedColors;
+  const borderColor = isSelected ? primaryColor : contrastLowColor;
 
   return isDisabled
     ? {
-        backgroundColor,
-        buttonColor: themedColors.disabledColor,
-        labelColor: themedColors.disabledColor,
+        buttonColor: disabledColor,
+        labelColor: disabledColor,
+        borderColor,
```

Comment:
Since half of the object is identical it might make sense to just have a condition around the part that is different.

Reply:
Fixed it